codec-haproxy/src/main/java/io/netty/contrib/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -185,27 +179,19 @@ static HAProxyMessage decodeHeader(ByteBuf header) {
                     "incomplete UNIX socket address information: " +
                             Math.min(addressInfoLen, header.readableBytes()) + " bytes (expected: 216+ bytes)");
             }
-            int startIdx = header.readerIndex();
-            int addressEnd = header.forEachByte(startIdx, 108, ByteProcessor.FIND_NUL);
-            if (addressEnd == -1) {
-                addressLen = 108;
-            } else {
-                addressLen = addressEnd - startIdx;
-            }
-            srcAddress = header.toString(startIdx, addressLen, CharsetUtil.US_ASCII);
+            int startIdx = header.readerOffset();
+            int bytes = header.openCursor(startIdx, 108).process(ByteProcessor.FIND_NUL);
+            addressLen = bytes == -1 ? 108 : bytes;
+            srcAddress = Statics.copyToCharSequence(header, startIdx, addressLen, CharsetUtil.US_ASCII).toString();
 
             startIdx += 108;
 
-            addressEnd = header.forEachByte(startIdx, 108, ByteProcessor.FIND_NUL);
-            if (addressEnd == -1) {
-                addressLen = 108;
-            } else {
-                addressLen = addressEnd - startIdx;
-            }
-            dstAddress = header.toString(startIdx, addressLen, CharsetUtil.US_ASCII);
+            bytes = header.openCursor(startIdx, 108).process(ByteProcessor.FIND_NUL);
+            addressLen = bytes == -1 ? 108 : bytes;
+            dstAddress = Statics.copyToCharSequence(header, startIdx, addressLen, CharsetUtil.US_ASCII).toString();
             // AF_UNIX defines that exactly 108 bytes are reserved for the address. The previous methods
             // did not increase the reader index although we already consumed the information.
-            header.readerIndex(startIdx + 108);
+            header.readerOffset(startIdx + 108);
```

Comment:
Once we have the length, we can still use `readCharSequence()` and then `skipReadable()` for what remains of the 108 reserved bytes.
That way we won't be relying on the internal `Statics`.